src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1559,6 +1583,30 @@ private void visitObjectType(FunctionType type, ObjectType prototype,
       emitBreak();
     }
 
+    /**
+     * Returns all interfaces implemented by a class and any
+     * superclasses for any of those interfaces.
+     */
+    public Iterable<ObjectType> getAllDirectlyImplementedInterfaces(FunctionType type) {
+      Set<ObjectType> interfaces = new HashSet<>();
+
+      for (ObjectType implementedInterface : type.getOwnImplementedInterfaces()) {
+        addRelatedInterfaces(implementedInterface, interfaces);
+      }
+      return interfaces;
+    }
+
+    private void addRelatedInterfaces(ObjectType instance, Set<ObjectType> interfaces) {
+      FunctionType constructor = instance.getConstructor();
+      if (constructor != null && constructor.isInterface() && !interfaces.contains(instance)) {
```

Comment:
`interfaces` is a set already, I don't see the point of checking `contains` before adding, just add it.